### PR TITLE
Sync changes from the rhds/old-main branch that Rishab created, so we can delete that branch

### PR DESCRIPTION
**Pull request by jiridanek**
I don't understand what it does. Do we want it? Maybe we should just delete old-main without merging it.

The description on the original commit was

> Perform update of the notebook controller without surge
>
> This allows updating the notebook controllers without increasing the
resources requirements of ODH during the update, which means it can
better tolerates situation such as the cluster being full.
>
> This means the notebook controller will be unavailable during the
update. However, this has not visible consequences except for a slight
delay in the notebooks CR reconciliation, until the new controller pod
start reconciling.

I guess it's quite plausible it has that effect. Do we still want it?

### CODE REVIEW

**Comment by atheo89** [comment]
This `main` branch looks old, and we didn't use it for long time (I guess this one was in sync with the v1.7 or older kbf release). I don't think that is a good idea to sync our main with this one.. Let's ping @harshad16 to add also his opinion on this.
 
The current `v1.9-branch` checked out from `https://github.com/opendatahub-io/kubeflow/tree/v1.7-branch` at the point

File: components/notebook-controller/config/manager/manager.yaml [hunk 1]
```diff
@@ -10,6 +10,10 @@ kind: Deployment
 metadata:
   name: deployment
 spec:
+  strategy:
```
Comment (atheo89) [inline components/notebook-controller/config/manager/manager.yaml:13]: That `spec.strategy`  option is already included on the managers' deployment files in both branches `main` and `v1.9-branch` 

https://github.com/opendatahub-io/kubeflow/blob/main/components/notebook-controller/config/manager/manager.yaml#L13-L16

File: components/odh-notebook-controller/config/manager/manager.yaml [hunk 2]
```diff
@@ -6,6 +6,10 @@ metadata:
   namespace: system
 spec:
   replicas: 1
+  strategy:
```
Comment (atheo89) [inline components/odh-notebook-controller/config/manager/manager.yaml:9]: The same for on https://github.com/opendatahub-io/kubeflow/blob/v1.9-branch/components/odh-notebook-controller/config/manager/manager.yaml#L9-L12

**Review by atheo89 — COMMENTED** [review]
Hey Jiri, looks like that these changes are exist on `main` and `v1.9-branch`.
So need to close this PR, however it is strange that your sync spot them..

**Comment by jiridanek** [comment]
@atheo89 thanks! I trusted the gh diff too much, thought that when it shows these lines as addition, they have to be new.

**Pull request closed without merging**